go/private/common.bzl

```diff
@@ -68,3 +68,29 @@ def pkg_dir(workspace_root, package_name):
   if package_name:
     return package_name
   return "."
+
+def dict_of(st):
+  """Converts struct objects into dictionaries."""
+  data = dict()
+  for key in dir(st):
+    value = getattr(st, key, None)
+    if value != None: # skip methods
+      data[key] = value
+  return data
+
+def merge(a, b):
```

Comment:
It seems strange for `merge` and `replace` to accept structs and return dicts. What do you think about taking a constructor as another argument and returning a struct?